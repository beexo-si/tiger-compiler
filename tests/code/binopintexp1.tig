/*TIGER_ARGS=-code*/
let
    var s:=2
    var x:=34
    var y:= s+x
in
    y
end
