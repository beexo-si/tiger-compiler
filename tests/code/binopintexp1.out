bien!
;;--FRAME--L0:
LABEL: L2
OPER: mov %d0, 2 D:[T3] S:[]
MOVE: mov %d0, %s0 D:T0 S:T3
OPER: mov %d0, 34 D:[T4] S:[]
MOVE: mov %d0, %s0 D:T1 S:T4
MOVE: mov %d0, %s0 D:T5 S:T0
OPER: add %d0, %s1 D:[T5] S:[T5,T1]
MOVE: mov %d0, %s0 D:T2 S:T5
MOVE: mov %d0, %s0 D:eax S:T2
OPER: jmp %j0 D:[] S:[] J:[L1]
LABEL: L1
OPER:  D:[] S:[eax,esp,ebp,ebx,edi,esi]
;;-END-FRAME-:
yes!!
